src/decider/network_decider/co_badged_card_info.rs

```diff
@@ -23,19 +22,44 @@ impl CoBadgedCardInfoList {
         }
     }
 
+    pub fn is_only_one_global_network_present(&self) -> bool {
+        let global_network_count = self
+            .0
+            .iter()
+            .filter(|card| card.card_network.is_global_network())
+            .count();
+
+        if global_network_count == 1 {
+            logger::debug!("Exactly one global network present");
+            true
+        } else {
+            logger::debug!(
+                "More than one global network present: count = {}",
+                global_network_count
+            );
+            false
+        }
```

Comment:
```suggestion
self
            .0
            .iter()
            .filter(|card| card.card_network.is_global_network())
            .count() == 1
```
if the logs is not important, then we can skip it if it gets printed lot of times

Reply:
This log is required for debugging, and it will get printed only once